### PR TITLE
Add Finetuning LLMs on CPUs Blog Post

### PR DESCRIPTION
**Pull request by Evaan2001**
# Description

This PR:
1. Creates a web page for my blog on Fine-tunuing an LLM on CPUs using Bazel and NativeLink
2. Adds some new words to vale vocab 
3. Makes a minor formatting change to nativelink/web/platform/src/content/docs/docs/config/production-config.mdx that is also presented in PR #1743. This formatting change needs to be merged into production before I can create a new blog post. Since PR #1743 will probably get merged later, I had to commit that change in this PR as well to make all the CI tests pass.

## Type of change

- [ ] Documentation update

## How Has This Been Tested?

The content and design of this web page are heavily inspired by the web page I made for my full AI Customer Service Agent blog post, which has already received LGTMs from Marcus and Aaron, though the latter has not yet been added to production. Once the page development was complete and the page was examined via `rm -rd dist && bun preview`, everything looked as expected, and no errors were thrown.

- - -
This change is [Reviewable](https://reviewable.io/reviews/TraceMachina/nativelink/1748)

### CODE REVIEW

File: web/platform/src/content/posts/Finetune_LLM_On_CPU.md [hunk 1]
```diff
@@ -0,0 +1,1078 @@
...
+
+### Initial Setup
+
+First, let's create our monorepo and structure it. Let’s name our repository `finetune-monorepo`.
+1\. Create a directory called `finetune-monorepo` and open your IDE from within it. Also create a `setup.sh` script. Alternatively, you can use the following shell commands
+<div style="max-height: 250px; max-width: 50vw; overflow: auto;">
+
+```bash
+# First cd into your preferred location
+mkdir finetune-monorepo
```
Comment (MarcusSorealheis) [inline web/platform/src/content/posts/Finetune_LLM_On_CPU.md]: I don't think you need to refer to it as a monorepo as much as just a repository. It's a small pproject

**Review by MarcusSorealheis — CHANGES_REQUESTED** [review]

File: web/platform/src/content/posts/Finetune_LLM_On_CPU.md (hunk 1 above)
Comment (Evaan2001) [reply web/platform/src/content/posts/Finetune_LLM_On_CPU.md]: Made the change!

**Review by MarcusSorealheis — APPROVED** [review]
:lgtm: keep the crazy ideas coming.

> Reviewed 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/TraceMachina/nativelink/1748#-:-OPVlD2wEPf9eRO22IfC:bte4bkv)* status: 1 of 2 LGTMs obtained, and 2 of 3 files reviewed, and 1 discussions need to be resolved

**Review by MarcusSorealheis — APPROVED** [review]

**Comment by Evaan2001** [comment]
@MarcusSorealheis and @aaronmondal updated this PR to reflect the changes discussed. If this looks good to you, please merge TraceMachina/nativelink-blogs/pull/1 so that I can test the `git clone` I've added. This won't take me more than 5 mins, so it'll be nice if we merge this after that. 

Also @aaronmondal, regarding your comment 'I believe you can control this via flags (i.e. --test_output=all)' on my statement "Print statements that track progress (like "Starting to fine tune model" or "Exiting this function") behave differently in remote execution. Unlike local runs where these appear in real-time, remote testing collects all logs on the server and only displays them after test completion when control returns to the local machine"

I've been running/testing my code all this while with this flag. `--test_output=all` preserves all output but only shows them after the entire run is finished.

**Review by aaronmondal — CHANGES_REQUESTED** [review]
nit: The layout seems inconsistent. I.e. why do we need the `1\.` instead of just `1.` and why do we use bold fonts in the titles when they're already titles?

> Reviewed 1 of 2 files at r3, 2 of 2 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/TraceMachina/nativelink/1748#-:-OQFnPbfAShdtLrJBcqP:b8p5eg1)* status: 1 of 2 LGTMs obtained, and all files reviewed, and pending CI: Bazel Dev / macos-15, Cargo Dev / macos-15, Cargo Dev / ubuntu-24.04, Installation / macos-14, Installation / macos-15, Local / lre-rs / macos-15, NativeLink.com Cloud / Remote Cache / macos-15, Publish image, Publish nativelink-worker-init, Remote / lre-cc / xlarge-ubuntu-24.04, Remote / lre-rs / xlarge-ubuntu-24.04, Web Platform Deployment / macos-15, buildstream, windows-2022 / stable, and 10 discussions need to be resolved

___
*[`web/platform/src/content/posts/Finetune_LLM_On_CPU.md` line 26 at r5](https://reviewable.io/reviews/TraceMachina/nativelink/1748#-OQFlvbx6lWNZWcMOlN7:-OQFlvbx6lWNZWcMOlN8:bxszyqa) ([raw file](https://github.com/TraceMachina/nativelink/blob/7d545262e5916caf5e6c8115fcbffd5b952bbcb4/web/platform/src/content/posts/Finetune_LLM_On_CPU.md#L26)):*
> ```Markdown
> ### Prerequisites
> 
> 1\. Bazel ([installation instructions](https://bazel.build/install)). This demo uses Bazel 8.1.1
> ```

nit: The demo uses Bazel `8.2.1`. A more future proof way in general might be to just use `1. A recent version of Bazel ([installation...).`
___
*[`web/platform/src/content/posts/Finetune_LLM_On_CPU.md` line 50 at r5](https://reviewable.io/reviews/TraceMachina/nativelink/1748#-OQFmZ35D9dmpoptbhEK:-OQFmZ35D9dmpoptbhEL:bb7lc5i) ([raw file](https://github.com/TraceMachina/nativelink/blob/7d545262e5916caf5e6c8115fcbffd5b952bbcb4/web/platform/src/content/posts/Finetune_LLM_On_CPU.md#L50)):*
> ```Markdown
> 2\. `requirements.lock` - Ensures consistent Python dependencies across all environments<br>
> 3\. `.bazelrc` - Main Bazel configuration file setting global options for hermetic builds and remote execution<br>
> 4\. `MODULE.bazel` - Configures the project as a Bazel module, tells Bazel we'll need python, pip and CPU-only PyTorch, and manages external dependencies<br>
> ```

nit:



_Suggestion:_
```Markdown
Python, `pip`
```
___
*[`web/platform/src/content/posts/Finetune_LLM_On_CPU.md` line 75 at r5](https://reviewable.io/reviews/TraceMachina/nativelink/1748#-OQFn75m8F7xU4COANvn:-OQFn75m8F7xU4COANvo:bw7dy83) ([raw file](https://github.com/TraceMachina/nativelink/blob/7d545262e5916caf5e6c8115fcbffd5b952bbcb4/web/platform/src/content/posts/Finetune_LLM_On_CPU.md#L75)):*
> ```Markdown
> 
> 
> ### The NativeLink Difference:
> ```

nit: Headers probably shouldn't end in a colon.
___
*[`web/platform/src/content/posts/Finetune_LLM_On_CPU.md` line 77 at r5](https://reviewable.io/reviews/TraceMachina/nativelink/1748#-OQFnF8Y66kB5I13mQfn:-OQFnF8Y66kB5I13mQfo:undefined) ([raw file](https://github.com/TraceMachina/nativelink/blob/7d545262e5916caf5e6c8115fcbffd5b952bbcb4/web/platform/src/content/posts/Finetune_LLM_On_CPU.md#L77)):*
> ```Markdown
> ### The NativeLink Difference:
> 
> To demonstrate NativeLink’s efficacy, consistency, and reliability, we ran the same fine-tuning job on the CPU of an M1 Pro MacBook Pro, the free version of Google Colab on CPU, and [NativeLink](https://github.com/TraceMachina/nativelink),which is free and open-source. We executed the fine-tuning task 5 times and this is what we observed:
> ```





_Suggestion:_
```Markdown
, w
```
___
*[`web/platform/src/content/posts/Finetune_LLM_On_CPU.md` line 98 at r5](https://reviewable.io/reviews/TraceMachina/nativelink/1748#-OQFnLWp4Wxvek22aKmq:-OQFnLWq9WIOj9K_md_0:b-vop18b) ([raw file](https://github.com/TraceMachina/nativelink/blob/7d545262e5916caf5e6c8115fcbffd5b952bbcb4/web/platform/src/content/posts/Finetune_LLM_On_CPU.md#L98)):*
> ```Markdown
> For forward-thinking AI teams, this infrastructure stack represents a shift from the "bigger is better" hardware arms race toward thoughtful resource utilization. The competitive advantage increasingly belongs to those who can extract maximum value from available compute rather than those who deploy more powerful hardware.
> ```

nit: Duplicate newline
___
*[`web/platform/src/content/docs/docs/config/production-config.mdx` line 18 at r5](https://reviewable.io/reviews/TraceMachina/nativelink/1748#-OQFlbfJBCDZdglvozqQ:-OQFlbfJBCDZdglvozqR:b-4nc0n0) ([raw file](https://github.com/TraceMachina/nativelink/blob/7d545262e5916caf5e6c8115fcbffd5b952bbcb4/web/platform/src/content/docs/docs/config/production-config.mdx#L18)):*
> ```MDX
> To run NativeLink, you just pass the path to a single JSON5 Configuration file, such as:
> 
> ```bash
> ```

nit: This is a nice fix, but unrelated to this PR. Consider pulling it out into a separate mini fix PR.
___
*[`.github/styles/config/vocabularies/TraceMachina/accept.txt` line 19 at r5](https://reviewable.io/reviews/TraceMachina/nativelink/1748#-OQFl6j7BwVuEeASTCdI:-OQFl6j7BwVuEeASTCdJ:bvza8fe) ([raw file](https://github.com/TraceMachina/nativelink/blob/7d545262e5916caf5e6c8115fcbffd5b952bbcb4/.github/styles/config/vocabularies/TraceMachina/accept.txt#L19)):*
> ```help
> [Hh]ermeticity
> JDK
> json
> ```

This shouldn't be in here. JSON should alwasy be written in capital letters. If you need the file ending `.json` the corresponding text should be in backticks.
___
*[`.github/styles/config/vocabularies/TraceMachina/accept.txt` line 38 at r5](https://reviewable.io/reviews/TraceMachina/nativelink/1748#-OQFlNq-8pgrXNOq3_LG:-OQFlNq-8pgrXNOq3_LH:byxcjgc) ([raw file](https://github.com/TraceMachina/nativelink/blob/7d545262e5916caf5e6c8115fcbffd5b952bbcb4/.github/styles/config/vocabularies/TraceMachina/accept.txt#L38)):*
> ```help
> OSSF
> Reclient
> [Rr]osetta
> ```

Rosetta should always be capitalized.
___
*[`.github/styles/config/vocabularies/TraceMachina/accept.txt` line 39 at r5](https://reviewable.io/reviews/TraceMachina/nativelink/1748#-OQFlQj-1POsTEpFcp4l:-OQFlQj-1POsTEpFcp4m:b-9fi58i) ([raw file](https://github.com/TraceMachina/nativelink/blob/7d545262e5916caf5e6c8115fcbffd5b952bbcb4/.github/styles/config/vocabularies/TraceMachina/accept.txt#L39)):*
> ```help
> Reclient
> [Rr]osetta
> [Rr]epo
> ```

Write out `repository` instead.

**Review by Evaan2001 — COMMENTED** [review]
Without the backslash in `1\.`, the 1. doesn't show in bun preview (screenshot attached)

I'll remove the bold from the titles (just thought it looked better).

> *[Reviewable](https://reviewable.io/reviews/TraceMachina/nativelink/1748#-:-OQFp_Bm0wOAwlYcalYJ:b-y207ui)* status: 1 of 2 LGTMs obtained, and all files reviewed, and 10 discussions need to be resolved

___
*[`web/platform/src/content/docs/docs/config/production-config.mdx` line 18 at r5](https://reviewable.io/reviews/TraceMachina/nativelink/1748#-OQFlbfJBCDZdglvozqQ:-OQFqXRz8cpS2kSC1qqc:bqxfdpy) ([raw file](https://github.com/TraceMachina/nativelink/blob/7d545262e5916caf5e6c8115fcbffd5b952bbcb4/web/platform/src/content/docs/docs/config/production-config.mdx#L18)):*
<details><summary><i>Previously, aaronmondal (Aaron Siddhartha Mondal) wrote…</i></summary><blockquote>

nit: This is a nice fix, but unrelated to this PR. Consider pulling it out into a separate mini fix PR.
</blockquote></details>

This was causing vale-related errors barring me from committing changes.

**Review by aaronmondal — APPROVED** [review]
:lgtm:

> Reviewed 2 of 2 files at r6, all commit messages.
> Dismissed @MarcusSorealheis from a discussion.
> *[Reviewable](https://reviewable.io/reviews/TraceMachina/nativelink/1748#-:-OQG-OaX9Rv-QSvJMxpB:bnfp4nl)* status: 2 of 2 LGTMs obtained, and all files reviewed, and pending CI: Cargo Dev / macos-15, Local / lre-rs / macos-15, NativeLink.com Cloud / Remote Cache / macos-15, buildstream, and 1 discussions need to be resolved

**Review by aaronmondal — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/TraceMachina/nativelink/1748)* status: <span>:shipit:</span> complete! 2 of 2 LGTMs obtained, and all files reviewed
